src/main/java/frc/robot/commands/PositionRecorder.java

```diff
@@ -0,0 +1,119 @@
+import edu.wpi.first.networktables.EntryNotification;
+import edu.wpi.first.networktables.NetworkTableEntry;
+import edu.wpi.first.wpilibj.geometry.Pose2d;
+import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardTab;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.subsystems.DriveTrain;
+
+public class PositionRecorder extends CommandBase {
+  /**
+   * Creates a new PositionRecorder.
+   */
+  public boolean isRunning = false;
+
+  DriveTrain drivetrain;
+  PrintWriter writer;
+  long start;
+
+  String directoryName="position-recordings";
+
+  NetworkTableEntry directoryNameEntry;
+  NetworkTableEntry isRunningEntry;
+  
+  public PositionRecorder(DriveTrain drivetrain) {
+    // Use addRequirements() here to declare subsystem dependencies.
+    initSmartDashboard();
+    this.drivetrain = drivetrain;
+  }
+  public void initSmartDashboard(){
+    ShuffleboardTab tab =Shuffleboard.getTab("Position Recorder");
+    isRunningEntry=tab.add("Is running", isRunning).withWidget("Toggle Button").getEntry();
+    directoryNameEntry=tab.add("Directory Name", directoryName).getEntry();
+
+
+    isRunningEntry.addListener((EntryNotification e)-> setIsRunning(e.value.getBoolean()), EntryListenerFlags.kUpdate|EntryListenerFlags.kNew);
+
+    directoryNameEntry.addListener((EntryNotification e)-> directoryName=e.value.getString(), EntryListenerFlags.kUpdate|EntryListenerFlags.kNew);
+  }
+
+  public void setIsRunning(boolean running){
+    System.out.println("Set is running: "+running);
+    if(running!=isRunning){
+      if(running){
+        this.schedule();
+      }else{
+        this.cancel();
+      }
+    }
+    if(isRunningEntry.getBoolean(false)!=running){
+      isRunningEntry.setBoolean(running);
+    }
+  }
+  // Called when the command is initially scheduled.
+  @Override
+  public void initialize() {
+    isRunning = true;
+    writer=null;
+    System.out.println("Position Recorder Begin");
+    try {
+      String filename = new SimpleDateFormat("MM-dd_HH_mm_ss").format(new Date())+".csv";
+
+      new File(directoryName).mkdirs();
+
+      File f = new File(directoryName,filename);
+      f.createNewFile();
+      writer = new PrintWriter(f); // PrintWriter is buffered
+      writer.println("Elapsed milliseconds,x,y,rotation");
+
+      start=System.currentTimeMillis();
+
+      System.out.println("Writing to: "+f.getAbsolutePath());
+
+    } catch (IOException e) {
+      // TODO Auto-generated catch block
+      e.printStackTrace();
+    }
+  }
+
+  // Called every time the scheduler runs while the command is scheduled.
+  @Override
+  public void execute() {
+    if(writer==null)return;
+    Pose2d currentPosition=drivetrain.getPose();
+    writer.println((System.currentTimeMillis()-start)+","+currentPosition.getX()+","+currentPosition.getY()+","+currentPosition.getRotation().getRadians());
+  }
```

Comment:
This should be in a try{} block. We don't want the robot program to throw exceptions just because it can't write to the file.